### PR TITLE
Fixed issue #18191: Cannot use statCount in specific condition

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: add a surveyid checker : is set + survey exist + survey is active

### CODE REVIEW

File: application/core/plugins/statFunctions/countFunctions.php [hunk 1]
```diff
@@ -80,4 +80,25 @@ public static function statCount($qCode, $submitted = true)
         }
         return intval(SurveyDynamic::model($surveyId)->count($oCriteria));
     }
+
+    /**
+     * Get the current valid survey id : Need to exist and activated
+     * Sometimes ExpressionManager don't get a valid survey id @see https://bugs.limesurvey.org/view.php?id=18191
+     * @return false[interger
+     */
+    private static function getCurrentSurveyId()
```
Comment (gabrieljenik) [inline application/core/plugins/statFunctions/countFunctions.php:89]: This could be a core function?
`getExecutingSurveyId()`

**Review by gabrieljenik — APPROVED** [review]
Code OK

File: application/core/plugins/statFunctions/countFunctions.php (hunk 1 above)
Comment (Shnoulle) [reply application/core/plugins/statFunctions/countFunctions.php:89]: LimeSurvey API ? Good idea

**Comment by Shnoulle** [comment]
Shit conflict … i redo the fix …

**Comment by Shnoulle** [comment]
> This could be a core function? `getExecutingSurveyId()`

`LimeSurvey\PluginManager\LimesurveyApi->getCurrentSurveyid($onlyactivated = false);` added

**Review by gabrieljenik — APPROVED** [review]
Code looks good.

**Pull request closed without merging**